src/Components/SingleArticleCard.jsx

```diff
@@ -0,0 +1,48 @@
+import { getSingleArticle } from "../api";
+import { useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+import Loading from "./Loading";
+import Error from "./Error";
+
+function SingleArticleCard() {
+  const { article_id } = useParams();
+  const [singleArticle, setSingleArticle] = useState({});
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+    setIsError(false);
+
+    getSingleArticle(article_id)
+      .then((articleData) => {
+        setSingleArticle(articleData);
+        setIsLoading(false);
+      })
+      .catch((error) => {
+        setIsError(true);
+        setIsLoading(false);
+      });
+  }, [article_id]);
+
+  if (isLoading) return <Loading />;
+  if (isError) return <Error />;
```

Comment:
This error logic is not part of this ticket and also doesn't work yet. Please delete it until you get to the error handling ticket